cli/internal/cmdutil/cmdutil.go

```diff
@@ -0,0 +1,218 @@
+// Package cmdutil holds functionality to run turbo via cobra. That includes flag parsing and configuration
+// of components common to all subcommands
+package cmdutil
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mattn/go-isatty"
+	"github.com/mitchellh/cli"
+	"github.com/spf13/pflag"
+	"github.com/vercel/turborepo/cli/internal/client"
+	"github.com/vercel/turborepo/cli/internal/config"
+	"github.com/vercel/turborepo/cli/internal/fs"
+	"github.com/vercel/turborepo/cli/internal/turbopath"
+	"github.com/vercel/turborepo/cli/internal/ui"
+)
+
+const (
+	// _envLogLevel is the environment log level
+	_envLogLevel = "TURBO_LOG_LEVEL"
+)
+
+// isCI returns true if running in a CI/CD environment
+func isCI() bool {
+	return !isatty.IsTerminal(os.Stdout.Fd()) || os.Getenv("CI") != ""
+}
+
+// Helper is a struct used to hold configuration values passed via flag, env vars,
+// config files, etc. It is not intended for direct use by turbo commands, it drives
+// the creation of CmdBase, which is then used by the commands themselves.
+type Helper struct {
+	// TurboVersion is the version of turbo that is currently executing
+	TurboVersion string
+
+	// for UI
+	forceColor bool
+	noColor    bool
+	// for logging
+	verbosity int
+
+	rawRepoRoot string
+
+	clientOpts client.Opts
+
+	// UserConfigPath is the path to where we expect to find
+	// a user-specific config file, if one is present. Public
+	// to allow overrides in tests
+	UserConfigPath turbopath.AbsolutePath
+
+	cleanups []io.Closer
+}
+
+// RegisterCleanup saves a function to be run after turbo execution,
+// even if the command that runs returns an error
+func (h *Helper) RegisterCleanup(cleanup io.Closer) {
+	h.cleanups = append(h.cleanups, cleanup)
+}
+
+// Cleanup runs the register cleanup handlers. It requires the flags
+// to the root command so that it can construct a UI if necessary
+func (h *Helper) Cleanup(flags *pflag.FlagSet) {
+	var ui cli.Ui
+	for _, cleanup := range h.cleanups {
+		if err := cleanup.Close(); err != nil {
+			if ui == nil {
+				ui = h.getUI(flags)
+			}
+			ui.Warn(fmt.Sprintf("failed cleanup: %v", err))
+		}
+	}
+}
+
+func (h *Helper) getUI(flags *pflag.FlagSet) cli.Ui {
+	colorMode := ui.GetColorModeFromEnv()
+	if flags.Changed("no-color") && h.noColor {
+		colorMode = ui.ColorModeSuppressed
+	}
+	if flags.Changed("color") && h.forceColor {
+		colorMode = ui.ColorModeForced
+	}
+	return ui.BuildColoredUi(colorMode)
+}
```

Comment:
I just discovered that this library also implements a full command set up, not just some niceties around string formatting. We also semi-erratically use our wrapper around it in `cli/internal/ui/ui.go`.

We could stand to invest more time into this area, eventually.